plugins/browser-plugin-youtube-tracking/src/types.ts

```diff
@@ -1,94 +1,180 @@
-import { MediaPlayerEvent, YouTube } from './contexts';
-import { SnowplowEvent } from './snowplowEvents';
-import { YTPlayerEvent, YTState } from './constants';
+import type { startMediaTracking, MediaEventType } from '@snowplow/browser-plugin-media';
+import { LEGACY_EVENTS } from './options';
 
-export type EventGroup = (SnowplowEvent | YTPlayerEvent | YTState | string)[];
+export type EventGroup = 'AllEvents' | 'DefaultEvents';
+export type CapturableEvents = (MediaEventType | EventGroup | keyof typeof LEGACY_EVENTS)[];
 
-export interface TrackingOptions {
-  mediaId: string;
-  player?: YT.Player;
-  urlParameters?: UrlParameters;
-  captureEvents: EventGroup;
-  youtubeEvents: YTPlayerEvent[];
-  label?: string;
-  updateRate?: number;
-  progress?: {
-    boundaries: number[];
-    boundaryTimeoutIds: ReturnType<typeof setTimeout>[];
-  };
-}
-
-export interface MediaTrackingOptions {
-  boundaries?: number[];
-  captureEvents?: EventGroup;
-  label?: string;
-  updateRate?: number;
-}
-
-export interface SnowplowYouTubeData {
-  percent?: number;
-  [key: string]: number | undefined;
-}
-
-export interface MediaEventData {
-  schema: string;
-  data: MediaPlayerEvent;
-  context: MediaEntities[];
-}
+type MediaTrackingConfiguration = Parameters<typeof startMediaTracking>[0];
+type VideoLocation = YT.Player | HTMLIFrameElement | string;
 
-export interface MediaEntities {
-  schema: string;
-  data: YouTube | SnowplowYouTubeData | SnowplowMediaPlayer;
-}
+/**
+ * YouTubeTrackingPlugin configuration values.
+ *
+ * Similar to the base Media Tracking plugin configuration, but with backwards compatibility for older versions of this YouTubeTrackingPlugin.
+ */
+export type YouTubeMediaTrackingConfiguration = {
+  /**
+   * The DOM ID for the element that hosts the YouTube video player to bind to.
+   * You can also pass the DOM element or `YT.Player` instance directly.
+   */
+  id?: VideoLocation;
```

Comment:
Do you think it would make sense to rename this property to something more specific? In the other media plugins we use `id` for the media session id and it seems like a weird inconsistency.

In the Vimeo plugin (which is based on the new media one), we use `video` for the video element and `id` for the session id. In case we used `video` here, perhaps we could avoid nesting the `options`?

Reply:
I had a similar [comment in the other media PR](https://github.com/snowplow/snowplow-javascript-tracker/pull/1344#discussion_r1776905002) – we would be changing the meaning of the `id` property which previously identified the player and now the media session. Should we perhaps rename the `enableYouTubeTracking` and `disableYouTubeTracking` functions to `startYouTubeTracking` and `endYouTubeTracking` to make sure that users notice the change of the API. The Vimeo and media plugins have a similar API and would be nice to make it consistent across all our media plugins (likely at the expense of more difficult migration to v4).

Reply:
Sure, this makes sense. I was attempting to maintain API compatibility as much as possible with v3, hoping that migration to v4 wouldn't really extend beyond the initial tracker setup.

But I think alignment with the other v2 plugins is a better goal. The cludge with `options` is confusing anyway I think so I'm happy to see that go.

I've gone with an adapter from the old API to the new one, how is that do you think?

Reply:
That's great, nice touch with the adapter!